src/common/query/src/lib.rs

```diff
@@ -24,6 +24,8 @@ pub mod physical_plan;
 pub mod prelude;
 mod signature;
 
+pub type Plugins = anymap::Map<dyn core::any::Any + Send + Sync>;
```

Comment:
Consider move this to a more universal module liike `common/base`

Reply:
+1

Reply:
How about creating a new crate "plugin"? It's wired to place "plugin"s either in "common/query" or "common/base". "base" is where we normally place some commonly used basic data structures and algorithms.

Reply:
It's actually a basic data structure which be aliased to a scenario-connected name :joy: 

I'm afraid that a new crate to too heavy for an alias. Maybe put into `base` for now and move if the more operations are added?

Reply:
move definition to `common-base` for now. if more plugin-related structs are made, then we can create a new plugin crate 🤣